components/authz-service/storage/v2/postgres/postgres_test.go

```diff
@@ -3307,6 +3307,638 @@ func TestUpdatePolicy(t *testing.T) {
 	}
 }
 
+func TestCreateRule(t *testing.T) {
+	store, db, _ := setup(t)
+	defer db.close(t)
+	defer store.Close()
+	ctx := context.Background()
+
+	cases := map[string]func(*testing.T){
+		"when the project doesn't exist, return Err": func(t *testing.T) {
+			condition1, err := storage.NewCondition(storage.Node, "chef-server-1", storage.ChefServer, storage.Equals)
+			require.NoError(t, err)
+			rule, err := storage.NewRule("new-id-1", "project-not-found", "name", storage.Node, []storage.Condition{condition1})
+			require.NoError(t, err)
+			resp, err := store.CreateRule(ctx, &rule)
+			assert.Nil(t, resp)
+			assert.Error(t, storage_errors.ErrForeignKey, err)
+		},
+		"cannot use improper condition attributes for events": func(t *testing.T) {
+			_, err := storage.NewCondition(storage.Event, "chef-server-1", storage.ChefTag, storage.Equals)
+			assert.Error(t, err)
+		},
+		"creating a rule with no conditions returns an error": func(t *testing.T) {
+			_, err := storage.NewRule("new-id-1", "project-1", "name", storage.Node, []storage.Condition{})
```

Comment:
Based on the first test, it looks like this test could also fail because the project doesn't exist, correct?